src/Cache.cc

```diff
@@ -805,6 +814,53 @@ Cache::getRoomName(lmdb::txn &txn, lmdb::dbi &statesdb, lmdb::dbi &membersdb)
         return "Empty Room";
 }
 
+JoinRule
+Cache::getRoomJoinRules(lmdb::txn &txn, lmdb::dbi &statesdb)
+{
+        using namespace mtx::events;
+        using namespace mtx::events::state;
+
+        lmdb::val event;
+        bool res = lmdb::dbi_get(
+          txn, statesdb, lmdb::val(to_string(mtx::events::EventType::RoomJoinRules)), event);
+
+        if (res) {
+                try {
+                        StateEvent<JoinRules> msg = json::parse(std::string(event.data(), event.size()));
+                        return msg.content.join_rule;
+                }
+                catch (const json::exception &e) {
+                        qWarning() << QString::fromStdString(e.what());
+                }
+        }
+        return JoinRule::Knock;
+}
+
+bool
+Cache::getRoomGuestAccess(lmdb::txn &txn, lmdb::dbi &statesdb)
+{
+        using namespace mtx::events;
+        using namespace mtx::events::state;
+
+        lmdb::val event;
+        bool res = lmdb::dbi_get(
+          txn, statesdb, lmdb::val(to_string(mtx::events::EventType::RoomGuestAccess)), event);
+
+        if (res) {
+                try {
+                        StateEvent<GuestAccess> msg = json::parse(std::string(event.data(), event.size()));
+                        if (msg.content.guest_access == AccessState::CanJoin)
```

Comment:
You can just write is as `return msg.content.guest_access == AccessState::CanJoin`.